docs/clients/s3.md

```diff
@@ -67,3 +67,23 @@ This information is required by AWS, and cannot be guessed by AsyncAws.
 If `ContentLength` is absent, AsyncAws will read the output before sending the
 request which could have a performance impact.
 
+
+### Download files
+```
+$getObjectResult = $s3->GetObject([
+    'Bucket' => 'my-company-website',
+    'Key' => 'humans.txt',
+]);
+$fp = fopen('/path/to/file.txt', 'wb');
+fwrite($fp, $result->getBody()->getContentAsString());
+
+// Download a big file with an iterable
+$fp = fopen('/path/to/big_file.mkv', 'wb');
+$getObjectResult = $s3->GetObject([
+    'Bucket' => 'my-company-website',
+    'Key' => 'bunny.mkv',
+]);
+foreach ($result->getBody()->getChunks() as $chunk) {
```

Comment:
I'd rather use the stream to copy the content in a file

```php
$fp = fopen('/path/to/big_file.mkv', 'wb');
stream_copy_to_stream($f->getBody()->getContentAsResource(), $s);
```

The iterrator could be use when you want to perform business logic between chunks

```php
$fp = fopen('/path/to/big_file.mkv', 'wb');
foreach ($result->getBody()->getChunks() as $chunk) {
    fwrite($fp, $chunk);
    $progress->advance();
}
```